rackcity/views/power_views.py

```diff
@@ -596,30 +558,75 @@ def make_bcman_request(chassis, blade, power_command):
     return result, exit_status
 
 
-def get_chassis_power_request_parameters(data):
+def get_chassis_power_request_parameters(request):
+    data = JSONParser().parse(request)
     if ("chassis_id" not in data) or ("blade_slot" not in data):
-        raise ChassisPowerManagementException(
+        raise PowerManagementException(
             "Must specify 'chassis_id' and 'blade_slot' on chassis power request."
         )
     try:
         blade_slot = int(data["blade_slot"])
         chassis_id = int(data["chassis_id"])
     except ValueError:
-        raise ChassisPowerManagementException(
+        raise PowerManagementException(
             "Parameters 'chassis_id' and 'blade_slot' must be of type int."
         )
     try:
         chassis = Asset.objects.get(id=chassis_id)
     except ObjectDoesNotExist:
-        raise ChassisPowerManagementException(
-            "Chassis" + GenericFailure.DOES_NOT_EXIST.value
+        raise PowerManagementException("Chassis" + GenericFailure.DOES_NOT_EXIST.value)
+    if not user_has_power_permission(request.user, asset=chassis):
+        raise UserPowerPermissionException(
+            "User "
+            + request.user.username
+            + " does not have power permission and does not own asset with id="
+            + str(data["id"])
         )
-    if (not chassis.model.is_blade_chassis) or (chassis.model.vendor != "BMI"):
-        raise ChassisPowerManagementException(
-            "Power is only network controllable for blade chassis of vendor 'BMI'."
+    if not is_asset_power_controllable_by_bcman(chassis):
+        raise PowerManagementException(
+            "Power is only network controllable for blade chassis of vendor 'BMI' with hostname not in storage."
         )
```

Comment:
I've mostly been using the convention of ending Exception messages with a period and a space so they can be chained together, but if you know that these are never followed by anything then I'm fine with it

Reply:
Goes for all the exceptions you raise in this PR

Reply:
ok word will do that